### PR TITLE
Codebase: Make private service fields readonly

### LINKED ISSUE #11008: Replacing properties `private T Service { get; } = Ioc.Default.GetRequiredService<T>()` by `readonly` field
### Description

Across the source code, we can find properties of this kind :

`private T Service { get; } = Ioc.Default.GetRequiredService<T>()`

As of late, in order to do some small optimizations, we started to replace properties retrieving `IUserSettingsService` by `private readonly` fields instead. There are other types of settings that are fetched by the same process. We should replace them as well by fields of this format:

`private readonly T service = Ioc.Default.GetRequiredService<T>()` (Note that the name variable starts by a lowercase letter)

### Concerned code

- Whole solution.

### Gains

- A slightly more optimized code.
- A more coherent code.

### Requirements

- `private T Service { get; } = Ioc.Default.GetRequiredService<T>()` are made into `private readonly T service = Ioc.Default.GetRequiredService<T>()`
- Public properties for sharing this format should be modified to look like this : 
```
private readonly T service = Ioc.Default.GetRequiredService<T>();
public T Service 
{ 
    get => return service; 
}
```

### Comments

_No response_

### PR DESCRIPTION
**Pull request by hecksmosis**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #11008 (if #11009 is merged, as this doesn't modify UserSettingsService)

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Review by ferrariofilippo — COMMENTED** [review]
I think that private non-static fields should be prefixed by '_'. See `2. Naming notation for variables` of <a href="https://files.community/docs/contributing/code-style">code style guidelines</a>

**Review by QuaintMako — APPROVED** [review]

**Comment by QuaintMako** [comment]
> I think that private non-static fields should be prefixed by '_'. See `2. Naming notation for variables` of [code style guidelines](https://files.community/docs/contributing/code-style)

This I am not sure. This notation has not been enfored for a while on the project (at least for the few months I have been active here). If we decide to enforce it, it will ask for a real effort to refactor the whole project in this direction. A decision should be taken regarding this in a proper discussion.

**Comment by yaira2** [comment]
Can you resolve the merge conflicts?

**Comment by hecksmosis** [comment]
Done

**Review by yaira2 — APPROVED** [review]
LGTM

File: src/Files.App/Filesystem/FilesystemOperations/ShellFilesystemOperations.cs [hunk 1]
```diff
@@ -27,6 +27,8 @@ public class ShellFilesystemOperations : IFilesystemOperations
 
 		private FilesystemOperations filesystemOperations;
 
+		private RecycleBinHelpers recycleBinHelpers;
+
```
Comment (ferrariofilippo) [inline src/Files.App/Filesystem/FilesystemOperations/ShellFilesystemOperations.cs:31]: `RecycleBinHelpers` is now static

File: src/Files.App/Filesystem/FilesystemOperations/ShellFilesystemOperations.cs [hunk 2]
```diff
@@ -27,6 +27,8 @@ public class ShellFilesystemOperations : IFilesystemOperations
...
 		private IDialogService DialogService { get; } = Ioc.Default.GetRequiredService<IDialogService>();
```
Comment (ferrariofilippo) [inline src/Files.App/Filesystem/FilesystemOperations/ShellFilesystemOperations.cs:32]: This should be refactored

**Review by ferrariofilippo — CHANGES_REQUESTED** [review]
There are also other services to be changed:
- `IFileTagsSettingsService` of `FileTagsService`
- `IFileTagsSettingsService` of `AboutViewModel`

File: src/Files.App/Filesystem/FilesystemOperations/ShellFilesystemOperations.cs (hunk 1 above)
Comment (hecksmosis) [reply src/Files.App/Filesystem/FilesystemOperations/ShellFilesystemOperations.cs:31]: Oh true (doesn't notice even though he did it)

File: src/Files.App/Filesystem/FilesystemOperations/ShellFilesystemOperations.cs (hunk 2 above)
Comment (hecksmosis) [reply src/Files.App/Filesystem/FilesystemOperations/ShellFilesystemOperations.cs]: It has been removed because it's not used

**Comment by hecksmosis** [comment]
That should be everything (according to the quick Visual Studio search I did)

**Review by ferrariofilippo — APPROVED** [review]
LGTM

**Pull request merged**